### PR TITLE
MGMT-14651: Add the option to create multiple disks on vSphere platform installation

### PR DESCRIPTION
**Pull request by eliorerz**
/cc @eliorerz

### CODE REVIEW

**Comment by eliorerz** [comment]
/override ci/prow/e2e-metal-assisted-ipv4v6
/override ci/prow/e2e-metal-assisted-ipv6
/override ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node
/override ci/prow/e2e-metal-assisted-kube-api-net-suite
/override ci/prow/e2e-metal-assisted-none
/override ci/prow/e2e-metal-assisted-static-ip-suite

File: terraform_files/vsphere/main.tf [hunk 1]
```diff
@@ -100,11 +100,15 @@ resource "vsphere_virtual_machine" "master" {
...
 
-  disk {
-    label            = "master-${count.index}-disk-0"
-    size             = var.master_disk_size_gib
-    eagerly_scrub    = false
-    thin_provisioned = true
+  dynamic "disk" {
+    for_each = range(var.master_disk_count)
+    content {
+      label            = "master-${count.index}-disk-${count.index + 1 + disk.key}"
```
Comment (osherdp) [inline terraform_files/vsphere/main.tf:106]: what's the idea of ``disk-${count.index + 1 + disk.key}``? why not say ``disk-${count.index + 1}-${disk.key}``?
also couldn't understand what's ``disk.key`` and where it's coming from

Comment (eliorerz) [reply terraform_files/vsphere/main.tf:106]: > what's disk.key and where it's coming from

https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/resources/virtual_machine#key

Comment (eliorerz) [reply terraform_files/vsphere/main.tf:106]: > why not say disk-${count.index + 1}-${disk.key}

I'll change it, that sounds more reasonable

Comment (eliorerz) [reply terraform_files/vsphere/main.tf:106]: OK, so the label for `disk-${count.index + 1}-${disk.key}` will be `master-0-disk-1-1` for disk number 2 while on `disk-${count.index + 1 + disk.key}` the label will be `master-0-disk-2`.

Comment (osherdp) [reply terraform_files/vsphere/main.tf:106]: yeah, I understand what's the difference between the options, but I think the addition of the numbers doesn't make sense and might lead to duplicated values (which is solved by having them separately)

**Review by osherdp — APPROVED** [review]

**Pull request merged**